cli/internal/taskhash/taskhash.go

```diff
@@ -129,45 +99,51 @@ func (th *Tracker) CalculateFileHashes(
 			return fmt.Errorf("missing pipeline entry %v", taskID)
 		}
 
-		pfs := &packageFileSpec{
-			pkg:    pkgName,
-			inputs: taskDefinition.Inputs,
+		pfs := &packageFileHashInputs{
+			taskID,
+			taskDefinition,
+			packageName,
 		}
 
 		hashTasks.Add(pfs)
 	}
 
-	hashes := make(map[packageFileHashKey]string, len(hashTasks))
-	hashObjects := make(map[packageFileHashKey]map[turbopath.AnchoredUnixPath]string, len(hashTasks))
-	hashQueue := make(chan *packageFileSpec, workerCount)
+	hashes := make(map[string]string, len(hashTasks))
+	hashObjects := make(map[string]map[turbopath.AnchoredUnixPath]string, len(hashTasks))
+	hashQueue := make(chan *packageFileHashInputs, workerCount)
 	hashErrs := &errgroup.Group{}
 
 	for i := 0; i < workerCount; i++ {
 		hashErrs.Go(func() error {
-			for packageFileSpec := range hashQueue {
-				pkg, ok := workspaceInfos.PackageJSONs[packageFileSpec.pkg]
+			for packageFileHashInputs := range hashQueue {
+				pkg, ok := workspaceInfos.PackageJSONs[packageFileHashInputs.packageName]
 				if !ok {
-					return fmt.Errorf("cannot find package %v", packageFileSpec.pkg)
+					return fmt.Errorf("cannot find package %v", packageFileHashInputs.packageName)
 				}
-				hashObject, err := packageFileSpec.getHashObject(pkg, repoRoot)
+
+				// Get the hashes of each file, keyed by the path.
+				hashObject, err := hashing.GetPackageFileHashes(repoRoot, pkg.Dir, packageFileHashInputs.taskDefinition.Inputs)
```

Comment:
Intentionally keeps the same signature that everywhere else does and doesn't redirect it so that we don't accidentally end up in the "N ways to do things" situation again.